testing/scripts/runtests

```diff
@@ -1,5 +1,13 @@
 #!/bin/sh -u
+export PATH=$PATH:~/.local/bin
 export CLASSPATH=.:$(pwd)/data/javatools.jar
-mkdir -p dist/test
-cabal build
-cabal exec -- cabal run -- --xml=dist/test/bnfc-system-tests.xml $*
+cd ../source
+stack setup
+stack init
+stack test
+stack install
```

Comment:
I totally agree that tests should not have such a side effect. Let me give you a bit more context here: When developing on bnfc I felt like installing everything needed to run the system tests locally in order to make sure that I don't break anything. On the other hand I did not want to install a lot of new tools only for running system tests (especially haskell-platform). This is why I came up with a docker-based solution to run the system tests locally in a lightweight virtual machine (ubuntu) without installing all the dependencies on my main system. You can we my first steps using docker in this [https://github.com/pascalh/bnfc/tree/docker](branch). The cool thing about the docker based approach could be that the CI also just runs the docker image and thus there should be no difference between the local and the CI system tests. If you are interested in using that approach for system tests, I can also add a pull request for it.